sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -20,12 +29,54 @@
 from ...model_selection import train_test_split
 from ...preprocessing import LabelEncoder
 from ._gradient_boosting import _update_raw_predictions
-from .common import Y_DTYPE, X_DTYPE, X_BINNED_DTYPE
+from .common import Y_DTYPE, X_DTYPE, X_BINNED_DTYPE, G_H_DTYPE
 
 from .binning import _BinMapper
 from .grower import TreeGrower
-from .loss import _LOSSES
-from .loss import BaseLoss
+
+
+_LOSSES = _LOSSES.copy()
+# TODO: Remove least_squares and least_absolute_deviation in v1.2
+_LOSSES.update(
+    {
+        "least_squares": HalfSquaredError,
+        "least_absolute_deviation": AbsoluteError,
+        "poisson": HalfPoissonLoss,
+        "binary_crossentropy": HalfBinomialLoss,
+        "categorical_crossentropy": HalfMultinomialLoss,
+    }
+)
+
+
+def _update_leaves_values(loss, grower, y_true, raw_prediction, sample_weight):
+    """Update the leaf values to be predicted by the tree.
+
+    Update equals:
+        loss.fit_intercept_only(y_true - raw_predictions)
+
+    This is only applied if loss.need_update_leaves_values is True.
+    Note: It only works, if the loss is a function of the residual, as is the
+    case for AbsoluteError and PinballLoss. Otherwise, one would need to get
+    the minimum of loss(y_true, raw_prediction + x) in x. A few examples:
+      - AbsoluteError: median(y_true - raw_predictions).
+      - PinballLoss: quantile(y_true - raw_predictions).
+    See also notes about need_update_leaves_values in BaseLoss.
+    """
+    # TODO: Ideally this should be computed in parallel over the leaves using something
+    # similar to _update_raw_predictions(), but this requires a cython version of
+    # median().
```

Comment:
I prefer to focus on the loss migration in this PR since this comment and implementation is in `main`:

https://github.com/scikit-learn/scikit-learn/blob/5d7dc4ba327c138cf63be5cd9238200037c1eb13/sklearn/ensemble/_hist_gradient_boosting/loss.py#L280-L282

We can work on improving the implementation in a follow up PR

Reply:
@jjerphan You could open an issue for your proposal so that we do not loose it.

Reply:
BTW, https://numpy.org/doc/stable/release/1.8.0-notes.html#performance-improvements-to-median. Not parallel, but already using partial sort.

Reply:
> @jjerphan You could open an issue for your proposal so that we do not loose it.

I opened https://github.com/scikit-learn/scikit-learn/issues/22173 accordingly.